### DISCUSSION TITLE
Rounding for matrices with different units

### DISCUSSION THREAD

**Discussion by kochjar** [josdejong/mathjs#3185] [Ideas]
I'm looking for a way to round elements within matrices that contain various units. Ideally, the function should retrieve the unit of the matrix entry at the specified index instead of requiring it to be explicitly provided in the third parameter.

```js
const example1 = math.round(math.evaluate("[[1.23],[4.56]]"),1)
const example2 = math.round(math.evaluate("[[1.23 cm],[4.56 km]]"), 1, math.unit("m"))
const notPossible = math.round(math.evaluate("[[1.23 s],[4.56 km]]"), 1)
```

Any suggestions or workarounds? Thanks!

**Comment by josdejong** [comment]
You can use `math.map` for that.

**Reply by kochjar** [reply]
Thank you! I didn't realize this existed. That's exactly what I needed.
